bugsnag-plugin-android-anr/src/main/jni/anr_handler.c

```diff
@@ -213,60 +207,93 @@ static void notify_anr_detected() {
   }
 }
 
-static void *sigquit_watchdog_thread_main(void *_) {
+static inline void block_sigquit() {
+  sigset_t sigmask;
+  sigemptyset(&sigmask);
+  sigaddset(&sigmask, SIGQUIT);
+  if (pthread_sigmask(SIG_BLOCK, &sigmask, NULL) != 0) {
+    BUGSNAG_LOG(
+        "Could not block SIGQUIT. Google's ANR handling will be disabled.");
+  }
+}
+
+static inline void unblock_sigquit() {
+  sigset_t anr_sigmask;
+  sigemptyset(&anr_sigmask);
+  sigaddset(&anr_sigmask, SIGQUIT);
+  if (pthread_sigmask(SIG_UNBLOCK, &anr_sigmask, NULL) != 0) {
+    BUGSNAG_LOG(
+        "Could not unblock SIGQUIT. Bugsnag's ANR handling will be disabled.");
+  }
+}
+
+static inline void trigger_sigquit_watchdog_thread() {
+  // Set the trigger flag for the fallback spin-lock in
+  // sigquit_watchdog_thread_main()
+  should_report_anr_flag = true;
+
+  if (should_wait_for_semaphore) {
+    // Although sem_post() is not officially marked as async-safe, the Android
+    // implementation simply does an atomic compare-and-exchange when there is
+    // only one thread waiting (which is the case here).
+    // https://cs.android.com/android/platform/superproject/+/master:bionic/libc/bionic/semaphore.cpp;l=289?q=sem_post&ss=android
+    if (sem_post(&anr_reporting_semaphore) != 0) {
+      // The only possible failure from sem_post is EOVERFLOW, which won't
+      // happen in this code. But just to be thorough...
+      BUGSNAG_LOG("Could not unlock Bugsnag sigquit handler semaphore");
+    }
+  }
+}
+
+static void *sigquit_watchdog_thread_main(__unused void *_) {
   static const useconds_t delay_2sec = 2000000;
   static const useconds_t delay_100ms = 100000;
   static const useconds_t delay_10ms = 10000;
```

Comment:
The 2s and 10ms delay constants are no longer used – do the latest changes make these unnecessary?